### PR TITLE
Generate OZ data

### PR DESCRIPTION
**Pull request by solovev**

### CODE REVIEW

File: packages/contracts/hardhat.config.ts [hunk 1]
```diff
@@ -23,15 +22,7 @@ import { ChainSymobls, getTokenBySymbol } from '@eonian/upgradeable'
...
-import './hardhat/tasks/start-hardhat-node.ts'
-
-task('accounts', 'Prints the list of accounts', async (taskArgs, hre) => {
-  const accounts = await hre.ethers.getSigners()
-
-  for (const account of accounts) {
-    console.log(account.address)
-  }
-})
+import './hardhat/tasks'
```
Comment (LeoVS09) [inline packages/contracts/hardhat.config.ts:25]: If it must be after all imports then maybe add comment?
```suggestion
// must be after all imports
import './hardhat/tasks'
```

File: packages/upgradeable/package.json [hunk 2]
```diff
@@ -56,6 +56,7 @@
     "@nomicfoundation/hardhat-network-helpers": "^1.0.9",
     "@nomiclabs/hardhat-ethers": "npm:hardhat-deploy-ethers@^0.3.0-beta.13",
     "@openzeppelin/hardhat-upgrades": "1.28.0",
+    "@openzeppelin/upgrades-core": "1.30.0",
```
Comment (LeoVS09) [inline packages/upgradeable/package.json:59]: Peer and dependencies? Maybe it must be in peer and dev dependencies?

File: packages/contracts/hardhat/tasks/generate-oz-manifest.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    entries.push({
+      contractName,
+      proxy,
+      implementation,
+    })
+  }
+  return entries
+}
+
+export const generateTask = task('generate-oz-manifest', 'Generates OpenZepellin manifest file (in ".openzepellin" folder)', action)
```
Comment (LeoVS09) [inline packages/contracts/hardhat/tasks/generate-oz-manifest.ts:41]: I not get, where this task is used? Does it included in deployment process?

File: packages/upgradeable/src/providers/validation.ts [hunk 4]
```diff
@@ -60,18 +140,15 @@ export class ValidationProvider {
...
    * @param deploymentName Name of the deployed contract. Consist from the actual contract name and tags.
    */
-  private async validateUpgrade(artifactName: string, deploymentName: string, options: StandaloneOptions) {
+  private async validateUpgrade(artifactName: string, deploymentName: string) {
     const deployment = await this.hre.deployments.getOrNull(deploymentName)
     if (!deployment) {
-      this.logger.log(`No deployment record found for "${deploymentName}". Has the contract never been deployed?`)
+      this.logger.debug(`No deployment record found for "${deploymentName}". Has the contract never been deployed?`)
       return
     }
```
Comment (LeoVS09) [inline packages/upgradeable/src/providers/validation.ts:148]: `validateDataInSync` and this method have same logic, maybe moe it in parent function?

**Review by LeoVS09 — APPROVED** [review]

File: packages/contracts/hardhat.config.ts (hunk 1 above)
Comment (solovev) [reply packages/contracts/hardhat.config.ts:25]: no, not necessarely, it should be below dotenv, but we already have the comment for it

File: packages/upgradeable/package.json (hunk 2 above)
Comment (solovev) [reply packages/upgradeable/package.json:59]: no, microbundle will inject it to the output bundle, that is not a desire behaviour

File: packages/contracts/hardhat/tasks/generate-oz-manifest.ts (hunk 3 above)
Comment (solovev) [reply packages/contracts/hardhat/tasks/generate-oz-manifest.ts:41]: i just needed some named var to export, to turn off TS errors

File: packages/upgradeable/src/providers/validation.ts (hunk 4 above)
Comment (solovev) [reply packages/upgradeable/src/providers/validation.ts:148]: this functions might be used separately later, so just for sanity check i'll leave it here (maybe i fix it later, once lib is done)

**Pull request merged**